frontend/src/pages/destructionlist/DestructionListItems.tsx

```diff
@@ -0,0 +1,142 @@
+import { AttributeData, DataGrid, Outline } from "@maykin-ui/admin-ui";
+import React, { useState } from "react";
+import {
+  useLoaderData,
+  useNavigation,
+  useSearchParams,
+  useSubmit,
+} from "react-router-dom";
+import { useAsync } from "react-use";
+
+import {
+  addToZaakSelection,
+  getZaakSelection,
+} from "../../lib/zaakSelection/zaakSelection";
+import { Zaak } from "../../types";
+import "./DestructionListDetail.css";
+import { DestructionListData, DestructionListDetailContext } from "./types";
+import { getFields, updateSelectedZaken } from "./utils";
+
+export type DestructionListItemsProps = {
+  zaken: Zaak[];
+  destructionList: DestructionListData;
+};
+
+export function DestructionListItems({
+  zaken,
+  destructionList,
+}: DestructionListItemsProps) {
+  const [isEditing, setIsEditing] = useState(false);
+  const { zaken: allZaken, zaaktypeChoices } =
+    useLoaderData() as DestructionListDetailContext;
+  const [searchParams, setSearchParams] = useSearchParams();
+  const { state } = useNavigation();
+  const submit = useSubmit();
+
+  useAsync(async () => {
+    await addToZaakSelection(String(destructionList.pk), zaken);
+  }, []);
+
+  const fields = getFields(searchParams, zaaktypeChoices);
+
+  const onUpdate = async () => {
+    const zaakSelection = await getZaakSelection(String(destructionList.pk));
+    const zaakUrls = Object.entries(zaakSelection)
+      .filter(([, selected]) => selected)
+      .map(([url]) => url);
+
+    const formData = new FormData();
+    zaakUrls.forEach((url) => formData.append("zaakUrls", url));
+
+    submit(formData, { method: "PATCH" });
+
+    setIsEditing(false);
+  };
+
+  const onFilter = (filterData: AttributeData) => {
+    const combinedParams = {
+      ...Object.fromEntries(searchParams),
+      ...filterData,
+    } as AttributeData<string>;
+
+    const activeParams = Object.fromEntries(
+      Object.entries(combinedParams).filter(
+        (keyValuePair) => !!keyValuePair[1],
+      ),
+    );
+
+    setSearchParams(activeParams);
+  };
+
+  const onSelect = async (
+    attributeData: AttributeData[],
+    selected: boolean,
+  ) => {
+    await updateSelectedZaken(
+      selected,
+      attributeData,
+      String(destructionList.pk),
+      zaken,
+    );
+  };
+
+  return (
+    <div className="zaken-list">
+      {isEditing ? (
+        <DataGrid
+          title="Zaakdossiers"
+          count={allZaken.count}
+          objectList={allZaken.results as unknown as AttributeData[]}
```

Comment:
Why is this type assertion (and similars in the render) needed? Don't we already type assert it higher up the tree at line 30? 
`const { zaken: allZaken, zaaktypeChoices } =
    useLoaderData() as DestructionListDetailContext;`

Reply:
In `DestructionListDetailContext` they are defined as `PaginatedZaken` and `allZaken.results` is then `Zaak[]`. I added it to make it stop complaining about it :S

Reply:
Hmm yeah, the `DataGrid` component expects `AttributeData[]` array I guess, a bit sad that these things aren't compatible, should be ok then haha